lib/src/arithmetic/fixed_to_float.dart

```diff
@@ -29,67 +29,106 @@ class FixedToFloat extends Module {
   /// Output port [float]
   late final FloatingPoint float = _float.clone()..gets(output('float'));
 
-  /// Constructor
+  /// Constructor for fixed to floating-point conversion.
+  /// - [fixed] point num ber to convert
+  /// - [exponentWidth] desired exponent width of output
+  /// - [mantissaWidth] desired mantissa width of output
+  /// - [signed] optional, if set to false (default true)
+  /// treat input as unsigned.
   FixedToFloat(FixedPoint fixed,
       {required this.exponentWidth,
       required this.mantissaWidth,
+      bool signed = true,
+      Logic? leadingDigitPredict,
       super.name = 'FixedToFloat'})
       : super(
             definitionName:
                 'Fixed${fixed.width}ToFloatE${exponentWidth}M$mantissaWidth') {
     fixed = fixed.clone()..gets(addInput('fixed', fixed, width: fixed.width));
     addOutput('float', width: _float.width) <= _float;
 
+    final localLeadingDigitPredict = (leadingDigitPredict != null)
```

Comment:
instead of creating a new "local" variable, why not just reassign `leadingDigitPredict = ...`?  You have no use for the original anymore during generation and reassigning will protect you from accidentally using it

Reply:
Indeed.  I modified the new code to reuse variables rather than creating new ones.  In this case, we are advancing both the jbit prediction and shifting left the guess on the mantissa.  Both of these are now reused to avoid incorrect use, similar to overwriting this input with the reassignment.

My confusion was that the type changes from Logic? to Logic and I didn't recognize this is legal.